DOCSUP-12212: Document currentRoles, enabledRoles, defaultRoles functions (#27170)
ClickHouse/ClickHouse: docs/en/sql-reference/functions/other-functions.md

-   [tcp_port](../../operations/server-configuration-parameters/settings.md#server_configuration_parameters-tcp_port)

## currentRoles {#current-roles}

> Returns the names of the roles which are current for the current user. The command [SET ROLE](../../sql-reference/statements/set-role.md#set-role-statement) could be used to change the current roles. If the command `SET ROLE` hasn't been used the function `currentRoles` returns the same as `defaultRoles`.

Reviewer: ```suggestion
Returns the names of the roles which are current for the current user. The current roles can be changed by the [SET ROLE](../../sql-reference/statements/set-role.md#set-role-statement) statement. If the `SET ROLE` statement was not used, the function `currentRoles` returns the same as `defaultRoles`.
```

> Returns the names of the roles which are current for the current user. The command [SET ROLE](../../sql-reference/statements/set-role.md#set-role-statement) could be used to change the current roles, by default the current roles are returned by the function [defaultRoles](#default-roles) called when the current user logins.

Reviewer 2: Instead of `by default the current roles are returned by the function [defaultRoles](#default-roles) called when the current user logins` please write `if the command `SET ROLE` hasn't been used the function currentRoles() returns the same as `defaultRoles()`.

**Syntax**

``` sql
currentRoles()
```

**Returned value**

-   List of the current roles for the current user. 

Type: [Array](../../sql-reference/data-types/array.md)([String](../../sql-reference/data-types/string.md)).

## enabledRoles {#enabled-roles}

> Returns the names of the current roles merged with those roles which are granted to some of the current roles.

Reviewer: ```suggestion
Returns the names of the current roles and the roles, granted to some of the current roles.
```

**Syntax**

``` sql
enabledRoles()
```

**Returned value**

-   List of the enabled roles for the current user. 

Type: [Array](../../sql-reference/data-types/array.md)([String](../../sql-reference/data-types/string.md)).

## defaultRoles {#default-roles}

> Returns the names of the roles which are set as current when the user logins. The command [SET DEFAULT ROLE](../../sql-reference/statements/set-role.md#set-default-role-statement) could be used to change the default roles. By default are all the roles which are granted to a user.

Reviewer 2: By default the default roles are all the roles which are granted to a user

Reviewer 2: Returns the names of the roles which are set as current when the current user logins.

> Returns the names of the roles which are set as current when the current user logins. The command [SET DEFAULT ROLE](../../sql-reference/statements/set-role.md#set-default-role-statement) could be used to change the default roles. By default the default roles are all the roles granted to the current user.

Reviewer: ```suggestion
Returns the names of the roles which are enabled by default for the current user when he logins. The current roles can be changed lately with the [SET ROLE](../../sql-reference/statements/set-role.md#set-role-statement) statement.
```
